code/modules/clothing/clothing_accessories.dm

```diff
@@ -1,21 +1,19 @@
 /obj/item/clothing/proc/can_attach_accessory(obj/item/clothing/accessory/A)
-	if(valid_accessory_slots && istype(A) && (A.slot in valid_accessory_slots))
+	if(src.valid_accessory_slots && (A.slot in src.valid_accessory_slots))
 		if(accessories.len && restricted_accessory_slots && (A.slot in restricted_accessory_slots))
 			for(var/obj/item/clothing/accessory/AC in accessories)
 				if (AC.slot == A.slot)
-					return 0
-		return 1
-	return 0
-
-	if(accessories.len && restricted_accessory_slots && (A.slot in restricted_accessory_slots))
-		for(var/obj/item/clothing/accessory/AC in accessories)
-			if (AC.slot == A.slot)
-				return 0
-	return 1
+					return FALSE
+			return TRUE
```

Comment:
Unnecessary line, since it's in the `if` statement, if it makes it past the `for` loop, it'll run into the `return TRUE` on the line immediately below when it gets out of this `if`